libraries/apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/internal/DefaultApolloStore.kt

```diff
@@ -36,7 +36,20 @@ internal class DefaultApolloStore(
 
   override val changedKeys = changedKeysEvents.asSharedFlow()
 
-  private val cache: OptimisticCache = OptimisticCache().chain(normalizedCacheFactory.createChain()) as OptimisticCache
+  // Initialize the cache lazily to avoid accessing the disk at initialization which usually happens on the main thread
+  private var _cache: OptimisticCache? = null
+  private val cache: OptimisticCache
+    get() {
+      if (_cache == null) {
+        lock.write {
+          if (_cache == null) {
+            _cache = OptimisticCache().chain(normalizedCacheFactory.createChain()) as OptimisticCache
+          }
+        }
+      }
```

Comment:
That doesn't work though. The whole `if {}` needs to be locked or you could create 2 caches

Reply:
Ah never mind, there's double if !